waltid-services/waltid-verifier-api/src/main/kotlin/id/walt/verifier/oidc/VerificationUseCase.kt

```diff
@@ -111,12 +111,13 @@ class VerificationUseCase(
         session
     }
 
-    fun getSession(sessionId: String): PresentationSession = sessionId.let { OIDCVerifierService.getSession(it) }!!
+    fun getSession(sessionId: String): PresentationSession = sessionId.let { OIDCVerifierService.getSession(it) }
+        ?: throw NotFoundException("State parameter doesn't refer to an existing session, or session expired: $sessionId")
 
     fun verify(sessionId: String?, tokenResponseParameters: Map<String, List<String>>): Result<String> {
         logger.debug { "Verifying session $sessionId" }
         val session = sessionId?.let { OIDCVerifierService.getSession(it) }
-            ?: return Result.failure(Exception("State parameter doesn't refer to an existing session, or session expired"))
+            ?: return Result.failure(NotFoundException("State parameter doesn't refer to an existing session, or session expired"))
```

Comment:
"State parameter $sessionId doesn't refer to an existing session or session is expired."